File: uplift-gradle-plugins/src/main/java/com/github/kjetilv/uplift/plugins/UpliftDestroyTask.java
package com.github.kjetilv.uplift.plugins;

import com.github.kjetilv.uplift.plugins.core.FileIO;
import org.gradle.work.DisableCachingByDefault;

/** Destroys the stack, then removes the generated CDK application. */
@DisableCachingByDefault(because = "Destroying a stack is not cacheable")
public abstract class UpliftDestroyTask extends UpliftCdkTask {

    @Override
    protected void perform() {
        if (!FileIO.isActualDirectory(cdkApp())) {
            initCdkApp();
        }
        cdk().run("cdk destroy --require-approval=never " + profileOption() + " " + getStack().get());
        FileIO.clearRecursive(cdkApp());
    }
}

File: uplift-gradle-plugins/src/main/kotlin/com/github/kjetilv/uplift/plugins/UpliftDestroyTask.kt
package com.github.kjetilv.uplift.plugins

import org.gradle.process.ExecOperations
import org.gradle.work.DisableCachingByDefault
import javax.inject.Inject

@DisableCachingByDefault(because = "Destroying a stack is not cacheable")
abstract class UpliftDestroyTask @Inject constructor(execOperations: ExecOperations) : UpliftCdkTask(execOperations) {

    override fun perform() {
        if (!cdkApp().isActualDirectory) {
            initCdkApp()
        }
        runCdk("cdk destroy --require-approval=never ${profileOption()} ${stack.get()}")
        clearCdkApp()
    }
}
